src/Maths/src/Cube.cs

```diff
@@ -0,0 +1,171 @@
+﻿using System;
+
+namespace Silk.NET.Maths
+{
+    /// <summary>
+    /// A structure representing a Cube with an <see cref="Origin"/> and <see cref="Size"/>
+    /// </summary>
+    public struct Cube<T>
+        : IEquatable<Cube<T>>
+        where T : unmanaged, IFormattable, IEquatable<T>, IComparable<T>
+    {
+        /// <summary>
+        /// The origin.
+        /// </summary>
+        public Vector3<T> Origin;
+        /// <summary>
+        /// The size.
+        /// </summary>
+        public Vector3<T> Size;
+
+        /// <summary>
+        /// Constructs a Cube from an origin and it's size
+        /// </summary>
+        /// <param name="origin">The origin of the cube.</param>
+        /// <param name="size">The size of the cube.</param>
+        public Cube(Vector3<T> origin, Vector3<T> size)
+        {
+            Origin = origin;
+            Size = size;
+        }
+        
+        /// <summary>
+        /// The center of this cube.
+        /// </summary>
+        public Vector3<T> Center
+        {
+            get => Origin + HalfSize;
```

Comment:
Can't be a property due to serializers getting thrown off by them.
Please convert to a method, the method version may share the same name for less typing.